### PR TITLE
Fix error with dragged text inside editor

### LINKED ISSUE #724: Editor crashes after performing drag & drop on styled text
## Are you reporting a feature request or a bug?

Bug

## Provide detailed reproduction steps (if any)

1. Put the caret at the beginning of the paragraph.
2. Press <kbd>Ctrl</kbd> + <kbd>B</kbd> and type `test`.
3. Select this word and drag it downwards.

### Actual result

Editor crashes.

```
ckeditor.js:160 Uncaught TypeError: Cannot read property 'type' of null
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.dom.range.setStart (ckeditor.js:160)
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.dom.range.setStartAfter (ckeditor.js:161)
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.dom.range.setStartAt (ckeditor.js:162)
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.dom.range.moveToPosition (ckeditor.js:159)
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.dom.range.splitElement (ckeditor.js:165)
    at ckeditor.js:372
    at $.insertHtml (ckeditor.js:340)
    at $.<anonymous> (ckeditor.js:350)
    at a.h (ckeditor.js:11)
    at a.window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.event.CKEDITOR.event.fire (ckeditor.js:12)
```

## GIF
bug_cke4

## Other details
Bug doesn't occur if there is the empty paragraph under dragged word.

* Browser: All browsers
* OS: Windows 10
* CKEditor version: 4.7.1

### PR DESCRIPTION
**Pull request by msamsel**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?
* Fix error when dragged text land within self. What result error with removed bookmarks during this operation.

Close #724

### CODE REVIEW

File: tests/plugins/clipboard/manual/draganddroperror.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@bender-tags: bug, 4.10.1, 724
+@bender-ui: collapsed
+@bender-ckeditor-plugins: wysiwygarea, toolbar, undo, sourcearea, elementspath, clipboard, basicstyles
+
+## Please note tat bug exist on Windows only, that's why you should perform a test under Windows, if it's possible.
```
[the remark this answers is not in the document]
Comment (msamsel) [reply tests/plugins/clipboard/manual/draganddroperror.md]: @Comandeer cna you explain how to to that? I'm not able to drag and move selected text on Chrome under macOS.

**Comment by msamsel** [comment]
> The fix is not working correctly in Edge. In such case there is visible lag after drop and before reselecting the dropped text. I was even able to hang the browser for several seconds before getting "NotSupportedError" and disappearing dropped text.

You have right that appear error in console, however reported issue mention about issue with error message:
`ckeditor.js:160 Uncaught TypeError: Cannot read property 'type' of null`.
Also code line reference seems to have source in completely different place, then reported. That's why I suspect, that error in Edge is not related to fixed part, and should be covered by separate issue. What do you think about that?

**Comment by Comandeer** [comment]
As we're working also on #1850, which is a bigger PR fixing drag&drop issues in Edge, it's probably wise to block this PR until #1850 is merged. Additionally, I'm changing this branch to target `major`, to merge into the same branch as mentioned Edge's PR.

**Comment by msamsel** [comment]
@Comandeer I've update this branch to be possible to merge.
As I can see #1850 is still not merged.
Current fix already patch issues in all browsers, however Edge is special. There is exposed another error, after fixing this one.
Do you still feel it's wise to block this PR, especially when #1850 is prolong in time, and we have ready fix for reported issue?

**Comment by f1ames** [comment]
#1850 seems to be quite difficult ticket. Maybe we could try to push this one without waiting for #1850? However, form the discussion above I'm not able to tell if this PR has broken something in Edge as suggested in https://github.com/ckeditor/ckeditor-dev/pull/2330#pullrequestreview-149670150 or not 🤔 

If this PR fixes the issue in all browsers apart from Edge and doesn't break Edge we should try to close it as #1850 can still take significant amount of time to finish. @Comandeer WDYT?

**Comment by Comandeer** [comment]
As #1850 is basically not fixable (every fix will break something), it seems that further waiting for it is really pointless. Therefore I think we could proceed with this PR.

**Comment by f1ames** [comment]
@Comandeer I can confirm that the https://github.com/ckeditor/ckeditor-dev/issues/724 issue is no longer reproducible and was fixed in #2803 as you mentioned. I suggest we just close this PR, WDYT?

**Comment by Comandeer** [comment]
Yup, as it's fixed there's no need to keep this PR open.

**Pull request closed without merging**